ida_settings/ida_settings.py

```diff
@@ -691,11 +701,11 @@ def user_plugin_names(self):
         return QtCore.QSettings(QtCore.QSettings.UserScope,
                                 IDA_SETTINGS_ORGANIZATION,
                                 IDA_SETTINGS_APPLICATION).childGroups()[:]
- 
+
     @classproperty
     def directory_plugin_names(self):
         ensure_ida_loaded()
-        return QtCore.QSettings(get_current_directory_config_path(),
+        return QtCore.QSettings(get_directory_config_path(directory=self._config_directory),
```

Comment:
this won't quite work. this property is meant to be a 'class property', that is, a property available from the class object itself, and not a specific instance. the intent is allow you to be able to query for all plugin names, without knowing any of them (and therefore, being unable to instantiate any `IDASettings`).